src/ir.cpp

```diff
@@ -20517,6 +20552,149 @@ static IrInstruction *ir_analyze_instruction_type_info(IrAnalyze *ira,
+                ConstExprValue *size_value = get_const_field(ira, payload, "size", 0);
+                assert(size_value->type == ir_type_info_get_type(ira, "Size", type_info_pointer_type));
+                uint32_t size_enum_index = bigint_as_u32(&size_value->data.x_enum_tag);
+                PtrLen ptr_len;
+                if (size_enum_index == 2) {
+                    ptr_len = PtrLenUnknown; // TODO: is this right?
+                } else {
+                    ptr_len = size_enum_index_to_ptr_len(size_enum_index);
+                }
+                ZigType *ptr_type = get_pointer_to_type_extra(ira->codegen,
+                    get_const_field_meta_type(ira, payload, "child", 4),
+                    get_const_field_bool(ira, payload, "is_const", 1),
+                    get_const_field_bool(ira, payload, "is_volatile", 2),
+                    ptr_len,
+                    bigint_as_u32(get_const_field_lit_int(ira, payload, "alignment", 3)),
+                    0, // bit_offset_in_host???
+                    0, // host_int_bytes???
+                    get_const_field_bool(ira, payload, "is_allowzero", 5)
+                );
+                if (size_enum_index != 2)
+                    return ptr_type;
+                return get_slice_type(ira->codegen, ptr_type);
+            }
+        case ZigTypeIdComptimeFloat:
+            return ira->codegen->builtin_types.entry_num_lit_float;
+        case ZigTypeIdComptimeInt:
+            return ira->codegen->builtin_types.entry_num_lit_int;
+        case ZigTypeIdUndefined:
+            return ira->codegen->builtin_types.entry_undef;
+        case ZigTypeIdNull:
+            return ira->codegen->builtin_types.entry_null;
+        case ZigTypeIdArray:
+        case ZigTypeIdStruct:
+        case ZigTypeIdOptional:
+        case ZigTypeIdErrorUnion:
+        case ZigTypeIdErrorSet:
+        case ZigTypeIdEnum:
+        case ZigTypeIdUnion:
+        case ZigTypeIdFn:
+        case ZigTypeIdBoundFn:
+        case ZigTypeIdArgTuple:
+        case ZigTypeIdOpaque:
+        case ZigTypeIdFnFrame:
+        case ZigTypeIdAnyFrame:
+        case ZigTypeIdVector:
+        case ZigTypeIdEnumLiteral:
+            ir_add_error(ira, instruction, buf_sprintf(
+                "@Type not implemented for 'TypeInfo.%s'\n", type_id_name(tagTypeId)));
+            return nullptr;
+    }
+    zig_unreachable();
+}
+
+static IrInstruction *ir_analyze_instruction_type(IrAnalyze *ira, IrInstructionType *instruction) {
+    IrInstruction *type_info_ir = instruction->type_info->child;
+    ConstExprValue *type_info_value = ir_resolve_const(ira, type_info_ir, UndefBad);
+    if (!type_info_value)
+        return ira->codegen->invalid_instruction;
+
+    // Do I need this assert?
+    assert(type_info_value->special == ConstValSpecialStatic);
+
+    // TODO: should we do the inverse of type_info_cache and/or re-use it here?
+
+    if (type_info_value->type != ir_type_info_get_type(ira, nullptr, nullptr)) {
```

Comment:
Instead of this check, I think it would be better to implicit cast `type_info_value` to this type. Look for calls to `ir_implicit_cast` to see some examples of this.